File: 1 lab/python/6.py
def multi(A, B):
    C = []
    for i in range(len(A)):
        C.append(A[i] * B[i])
    return sum(C)

def roll(A, n, m):
    res = []
    for j in range(m):
        temp = []
        for i in range(n):
            temp.append(A[i][j])
        res.append(temp)
    return res

n, m, k = map(int, input().split())
A = []
B = []
C = []
for i in range(n):
    A.append(list(map(int, input().split())))
for i in range(m):
    B.append(list(map(int, input().split())))

B = roll(B, m, k)

for a in range(n):
    temp = []
    for b in range(k):
        temp.append(multi(A[a], B[b]))
    C.append(temp)

for i in C:
    print(*i)

File: 1 lab/cpp/6.cpp
#include <iostream>
#include <numeric>
using namespace std;

int multi(vector<int> A, vector<int> B) {
    vector<int> C;
    for (int i = 0; i < A.size(); i++) {
        C.push_back(A[i] * B[i]);
    }
    return accumulate(C.begin(), C.end(), 0);
}

vector<vector<int>> roll(vector<vector<int>> A, int n, int m) {

    vector<vector<int>> res;
    vector<int> temp;

    for (int j = 0; j < m; j++) {
        temp.clear();
        for (int i = 0; i < n; i++) {
            temp.push_back(A[i][j]);
        }
        res.push_back(temp);
    }

    return res;
}

int main(void) {
    int n, m, k;
    cin >> n >> m >> k;
    vector<vector<int>> A(n, vector<int>(m));
    vector<vector<int>> B(m, vector<int>(k));
    vector<vector<int>> C;

    for (int i = 0; i < n; i++) {
        for (int j = 0; j < m; j++) {
            cin >> A[i][j];
        }
    }

    for (int i = 0; i < m; i++) {
        for (int j = 0; j < k; j++) {
            cin >> B[i][j];
        }
    }

    B = roll(B, m, k);

    vector<int> temp;
    for (int a = 0; a < n; a++) {
        temp.clear();
        for (int b = 0; b < k; b++) {
            temp.push_back(multi(A[a], B[b]));
        }
        C.push_back(temp);
    }

    for (vector<int> i : C) {
        for (int j = 0; j < i.size(); j++) {
            cout << i[j];
            if (j != i.size() - 1) cout << " ";
        }
        cout << endl;
    }

    return 0;
}